pkg/backends/websockets.go

```diff
@@ -47,21 +43,19 @@ func NewWebsocketDialer(address string, tlscfg *tls.Config, extraHeader string,
 func (b *WebsocketDialer) Start(bsf netceptor.BackendSessFunc, errf netceptor.ErrorFunc) {
 	go func() {
 		for {
-			cfg, err := websocket.NewConfig(b.address, b.origin)
-			if err != nil {
-				errf(fmt.Errorf("error creating websocket config"), true)
-				return
-			}
+			dialer := websocket.DefaultDialer
 			if b.tlscfg != nil {
-				cfg.TlsConfig = b.tlscfg
+				dialer.TLSClientConfig  = b.tlscfg
 			}
+			header := make(http.Header, 0)
```

Comment:
This declaration is going to be shadowed by the `:=` a few lines below, inside the `if {}` block.  This is a go-ism - every curly brace block implies a lexical scope, so the second `:=` declares a new variable that only exists within the `if {}`.  I think you get the behavior you want by just deleting the second declaration (line 60), which will make the `header.Add` applies to the outer variable.

Reply:
fixed